core/client/fs/src/main/java/alluxio/client/block/AlluxioBlockStore.java

```diff
@@ -298,6 +295,9 @@ public BlockOutStream getOutStream(long blockId, long blockSize, WorkerNetAddres
    */
   public BlockOutStream getOutStream(long blockId, long blockSize, OutStreamOptions options)
       throws IOException {
+    if (getEligibleWorkers().isEmpty()) {
```

Comment:
can we do this check when we know there are no locations? https://github.com/Alluxio/alluxio/blob/master/core/client/fs/src/main/java/alluxio/client/block/AlluxioBlockStore.java#L317